src/components/MultiHopTrade/components/TradeAmountInput.tsx

```diff
@@ -269,12 +270,12 @@ export const TradeAmountInput: React.FC<TradeAmountInputProps> = memo(
                 <Skeleton isLoaded={!showFiatSkeleton}>{oppositeCurrency}</Skeleton>
               </Button>
             )}
-            {onPercentOptionClick && (
+            {(onMaxClick || onPercentOptionClick) && (
               <PercentOptionsButtonGroup
                 options={percentOptions}
                 isDisabled={isReadOnly || isSendMaxDisabled}
                 onMaxClick={onMaxClick}
-                onClick={onPercentOptionClick}
+                onClick={onPercentOptionClick ?? noop}
```

Comment:
Since it's impossible to get the discriminated union right, and `onClick` should always be here as a base to avoid not passing it and getting rugged.